### ISSUE TITLE
security/acme-client: action to push a cert to Ruckus ZoneDirector or Unleashed

### ISSUE THREAD

**Issue by alaricljs** [opnsense/plugins#3536]
I'd like to see this capability added which is similar to other Acme actions, here's an example of a working method in pfSense: https://ms264556.net/pages/PfSenseLetsEncryptToRuckus

Tried doing this myself, but don't have the cycles to come up to speed on PHP, MVC, ...

**Comment by ms264556** [comment]
@alaricljs the initial step would be to add a deploy hook to [acme.sh](https://github.com/acmesh-official/acme.sh).

If you add a feature request to the acme.sh issues, and ping me the details, then I can look at doing a PR if I have some spare time.

**Comment by Churator** [comment]
> @alaricljs the initial step would be to add a deploy hook to [acme.sh](https://github.com/acmesh-official/acme.sh).
> 
> If you add a feature request to the acme.sh issues, and ping me the details, then I can look at doing a PR if I have some spare time.

https://github.com/acmesh-official/acme.sh/issues/5035

**Comment by uberjay** [comment]
There's an open PR which adds support for deploying to Ruckus Unleashed. I tested it by hand on my OPNsense deployment and it worked perfectly. I posted [a comment over there](https://github.com/acmesh-official/acme.sh/pull/4832#issuecomment-2427599297) with the details.

Unfortunately the PR has been open for quite some time with no response, so I'm not really hopeful for forward progress, other than what I've already done!

**Comment by uberjay** [comment]
Good news -- the upstream PR is merged into the acme.sh dev branch. I'm not totally familiar with how their dev branch changes flow into a release, but I guess it'll happen soon enough. After that's in, _then_ this plugin can get updated to include the deploy hook option?

**Comment by ms264556** [comment]
> @ms264556 That's correct. 👍 When the functionality was added to acme.sh, just ping me here and I'll reopen this issue (it will most likely be closed due to timeout).

Hi @fraenki. The ruckus deploy book has landed in acme.sh master now, so you can re-open this issue. Thanks!

**Comment by fraenki** [comment]
> The ruckus deploy hook has landed in acme.sh master now

Great! Unfortunately it did not make it into the acme.sh 3.1.0 release. So we'll have to wait for the next release of acme.sh, which will most likely include the "ruckus" deploy hook.

Note to myself: [link to the hook documentation](https://github.com/acmesh-official/acme.sh/wiki/deployhooks#34-deploy-your-certificate-to-ruckus-unleashed-or-ruckus-zonedirector).

**Comment by jeffmaki** [comment]
Just a user report here: 

First, thanks for putting this together. I found it when searching for a way to upgrade my Unleashed with LE certs, so great minds think alike.

The "but" here is that this script does not do any validation against the requirements of the Unleashed portal. I ended up uploading ECC certs (dumb in retrospect) to an old R710 and it bricked the SSL admin portal. Strangely, even a factory reset of the unit doesn't clear these certs, which left me in a real bind. 

I ended up having to get clever and using an exploit (https://alephsecurity.com/vulns/aleph-2019004) to get access to the shell and copying the original self-signed certs back to access the SSL portal and upload proper LE certs using this script.

FWIW, I ended up downloading the CSR from the portal and using that to get the LE certs vs. replacing the private key, just to be safe. I also had to change the private key length to 2048 since 1024 bit keys are no longer supported in 2025. (This is again old hardware, R710). 

So TL;DR: all's good now, I have this script working properly, but it's super easy--perhaps too easy--for somebody to shoot themselves in the foot with this. Be warned. Hope this helps somebody if they also get stuck like I did.

**Comment by ms264556** [comment]
The [documentation calls out the RSA-only restriction](https://github.com/acmesh-official/acme.sh/wiki/deployhooks#35-deploy-your-certificate-to-ruckus-unleashed-or-ruckus-zonedirector), but the deployhook code makes no checks.

I'll see how straightforward it is to add a check to the hook.

**Comment by ms264556** [comment]
The Ruckus deploy hook has made it into a tagged acme.sh release ([3.1.1](https://github.com/acmesh-official/acme.sh/releases/tag/3.1.1)).

**Comment by Churator** [comment]
When should we expect this to propagate to OpnSense ?
